### ISSUE TITLE
Psedo True Abundance Map Generation

### ISSUE THREAD

**Issue by Boinao** [danfenghong/IEEE_TNNLS_EGU-Net#4]
First thing first thanks for sharing the code and a wonderful work indeed.
I have one query regarding the True Abundance Map Generation (TrLabel)

Abund = sunsal(EM, X', 'lambda', 0, 'ADDONE', 'no', 'POSITIVITY', 'yes', ...
            'AL_iters', 200, 'TOL', 1e-4, 'verbose','yes');
TrLabel=(Abund./repmat(sum(Abund), size(Abund, 1), 1));

This will give an output of EM=8000x Channel (Pure_TrSet)
Abund here will give an output of  8000x Channel (TrLabel)

But the sample TrLabel has 8000x5

I am not really sure how the TrLabel is generated?
It will be very helpful if you can elaborate on this

Thanks once again

**Comment by danfenghong** [comment]
Hi,

Thank you very much for your interest in our work!
The TrLabel, i.e., Abundance maps, has 8000x5 size. Since we have five
materials, i.e., one-pixel spectral signature should be unmixed to be five
components, so This is why here is 8000*5.

The mechanism of unmixing can be well found in our paper.
Thanks again!
Cheers,
Danfeng

Boinao ***@***.***> 于2021年8月18日周三 下午10:30写道：

**Comment by Boinao** [comment]
I understand that it has to be 8000x5
but the code in Pseudo_endmembers_generation.m line # 23
Abund = sunsal(EM, X', 'lambda', 0, 'ADDONE', 'no', 'POSITIVITY', 'yes', ...
            'AL_iters', 200, 'TOL', 1e-4, 'verbose','yes');

EM here is around (8000xChannel) and X' (NxChannel) here is the data.
How using Sunsal true label is generated to be **8000x5**. This part is very confusing. 
The correct way should have been

Abund = sunsal(True_EM, EM, 'lambda', 0, 'ADDONE', 'no', 'POSITIVITY', 'yes', ...
            'AL_iters', 200, 'TOL', 1e-4, 'verbose','yes');

Would you please clarify for me.

Regards,

Boinao

**Comment by danfenghong** [comment]
Yes, yes. You are right. We made a mistake. We have updated the codes.
Sorry for that!
Thank you very much for pointing that out!
Cheers,
Danfeng

Boinao ***@***.***> 于2022年1月6日周四 22:18写道：

**Comment by jikerWRN** [comment]
> 
Hi, good question! I want to know how do you get the "True_EM"? Can you get good results? Looking forwars to your reply!

**Comment by danfenghong** [comment]
you can use some endmember extraction methods for EM, e.g., VCA. Their EMs
used are the same for those LMM-based methods.

jikerWRN ***@***.***> 于2022年1月17日周一 17:33写道：
